routing/payment_lifecycle.go

```diff
@@ -243,7 +243,13 @@ lifecycle:
 		// Now request a route to be used to create our HTLC attempt.
 		rt, err := p.requestRoute(ps)
 		if err != nil {
-			return exitWithErr(err)
+			log.Errorf("Failed to find a route for payment %v: %v",
+				p.identifier, err)
+
+			// The payment is now marked as failed, we'll continue
```

Comment:
I am not sure if this comment holds tho because in the requestRoute func, we would not necessarily fail the payment depending on the error:

```
	var routeErr noRouteError
	if !errors.As(err, &routeErr) {
		return nil, fmt.Errorf("requestRoute got: %w", err)
	}
```

Reply:
oh nice yeah it was handled there already, one less commit

Reply:
Not sure I think we need to keep this exception, imagine we launch a payment, we need to shard the payment, so we send one paymenet successfully now we try the next shard, run out of routes ? We would then close the payment_lifecycle and remove all the result collectors again ...

Reply:
if it's no route error we'd return a nil error and continue the lifecycle due to nil route below.

Reply:
hmm aren't we returning an error when it's no route err ?:

`return nil, fmt.Errorf("requestRoute got: %w", err)`

Reply:
in that case we also wanna wait for all payment to resolve tho.

Reply:
Good point, makes sense to exit. My only concern here is then that we are closing the result collectors, which is unfortunate.

Reply:
as long as the router and switch are active we probably should keep them alive.